test/e2e_node/pod_conditions_test.go

```diff
@@ -57,40 +57,65 @@ var _ = SIGDescribe("Pod conditions managed by Kubelet", func() {
 		ginkgo.It("a pod with init containers should report all conditions set in expected order after the pod is up", runPodReadyConditionsTest(f, true, false))
-		ginkgo.It("a pod failing to mount volumes and without init containers should report scheduled and initialized conditions set", runPodFailingConditionsTest(f, false, false))
-		ginkgo.It("a pod failing to mount volumes and with init containers should report just the scheduled condition set", runPodFailingConditionsTest(f, true, false))
+		ginkgo.It("a pod failing to mount volumes and without init containers should report scheduled and initialized conditions set", runPodFailingConditionsTest(f, false, false, "ConfigMap"))
+		ginkgo.It("a pod failing to mount volumes and with init containers should report just the scheduled condition set", runPodFailingConditionsTest(f, true, false, "ConfigMap"))
+		ginkgo.It("a pod failing to mount volumes (Secret) and without init containers should report scheduled and initialized conditions set", runPodFailingConditionsTest(f, false, false, "Secret"))
+		ginkgo.It("a pod failing to mount volumes (Secret) and with init containers should report just the scheduled condition set", runPodFailingConditionsTest(f, true, false, "Secret"))
 		addAfterEachForCleaningUpPods(f)
 	})
 })
 
-func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, checkPodReadyToStart bool) func(ctx context.Context) {
+func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, checkPodReadyToStart bool, volumeSource string) func(ctx context.Context) {
 	return func(ctx context.Context) {
 		ginkgo.By("creating a pod whose sandbox creation is blocked due to a missing volume")
 
 		p := webserverPodSpec("pod-"+string(uuid.NewUUID()), "web1", "init1", hasInitContainers)
-		p.Spec.Volumes = []v1.Volume{
-			{
-				Name: "cm",
-				VolumeSource: v1.VolumeSource{
-					ConfigMap: &v1.ConfigMapVolumeSource{
-						LocalObjectReference: v1.LocalObjectReference{Name: "does-not-exist"},
+
+		switch volumeSource {
+		case "ConfigMap":
+			p.Spec.Volumes = []v1.Volume{
+				{
+					Name: "cm",
+					VolumeSource: v1.VolumeSource{
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{Name: "does-not-exist-configmap"},
+						},
 					},
 				},
-			},
-		}
-		p.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
-			{
-				Name:      "cm",
-				MountPath: "/config",
-			},
+			}
+			p.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+				{
+					Name:      "cm",
+					MountPath: "/config"},
+			}
+		case "Secret":
+			p.Spec.Volumes = []v1.Volume{
+				{
+					Name: "secret",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName: "does-not-exist-secret",
+						},
+					},
+				},
+			}
+			p.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+				{
+					Name:      "secret",
+					MountPath: "/secret"},
```

Comment:
nit:

```suggestion
				{
					Name:      "secret",
					MountPath: "/secret",
				},
```

Reply:
addressed in https://github.com/kubernetes/kubernetes/commit/b06974f81f9622dfcbe919b2e57f4862963342dc